hack/tools/internal/tilt-prepare/main.go

```diff
@@ -871,6 +871,9 @@ func updateDeployment(prefix string, objs []unstructured.Unstructured, f updateD
 		if obj.GetKind() != "Deployment" {
 			continue
 		}
+		if _, exists := obj.GetLabels()[clusterv1.ProviderNameLabel]; !exists {
```

Comment:
Could you kindly add a comment why we are doing this, e.g. 
```suggestion
		// Ignore Deployments that are not part of the provider, eg. ASO in CAPZ.
		if _, exists := obj.GetLabels()[clusterv1.ProviderNameLabel]; !exists {
```